cmd/fs/helpers.go

```diff
@@ -46,6 +48,65 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(pathArgCount int, onlyDirs bool, filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error)) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClient(cmd, args) // TODO: Fix this
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isValidPrefix := isDbfsPath(toComplete)
+
+		if !isValidPrefix {
+			return []string{dbfsPrefix}, cobra.ShellCompDirectiveNoSpace
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		wsc := root.WorkspaceClient(cmd.Context())
+		_, err = wsc.CurrentUser.Me(cmd.Context())
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		completer := completer.NewCompleter(cmd.Context(), filer, onlyDirs)
+
+		if len(args) < pathArgCount {
+			completions, directive := completer.CompleteRemotePath(path)
+
+			// The completions will start with a "/", so we'll prefix
+			// them with the dbfsPrefix without a trailing "/" (dbfs:)
+			prefix := dbfsPrefix[:len(dbfsPrefix)-1]
+
+			// Add the prefix to the completions
+			for i, completion := range completions {
+				completions[i] = fmt.Sprintf("%s%s", prefix, completion)
+			}
+
+			return completions, directive
+		} else {
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+	}
+}
+
+// Wrapper for [root.MustWorkspaceClient] that disables loading authentication configuration from a bundle.
+func mustWorkspaceClient(cmd *cobra.Command, args []string) error {
+	if root.HasWorkspaceClient(cmd.Context()) {
+		return nil
+	}
```

Comment:
Is it ok to bail early here or is it possible we have a workspaceClient with configuration from a bundle?

Reply:
+1. The `fs` commands respect the bundle configuration. If we skip loading bundle auth below we could potentially be providing autocomplete suggestions from a different workspace.

Reply:
Chatted with @andrewnester and removed this whole method. `fs` autocomplete now respects the bundle configuration 👍